flocker/provision/_install.py

```diff
@@ -992,6 +995,7 @@ def configure_cluster(cluster, dataset_backend_configuration):
                             dataset_backend_configuration=(
                                 dataset_backend_configuration
                             ),
+                            public_ip=node.address if node._node.driver.name == 'Amazon EC2' else None,
```

Comment:
This should probably be changed. Perhaps it should just be passed unconditionally? (My first inclination was to not do that, so we test the case where it isn't specified).